File: rb/features.rb
# PollinationsAi SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module PollinationsAiFeatures
  def self.make_feature(name)
    case name
    when "base"
      PollinationsAiBaseFeature.new
    when "test"
      PollinationsAiTestFeature.new
    else
      PollinationsAiBaseFeature.new
    end
  end
end

File: py/features.py
# PollinationsAi SDK feature factory

from feature.base_feature import PollinationsAiBaseFeature
from feature.test_feature import PollinationsAiTestFeature


def _make_feature(name):
    features = {
        "base": lambda: PollinationsAiBaseFeature(),
        "test": lambda: PollinationsAiTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
